### PR TITLE
feat: Add proton calendar

### LINKED ISSUE #14884: [CAL-3640] App: Proton Calendar
add https://proton.me/calendar to https://app.cal.com/apps/categories/calendar with proper OAuth

waiting for them to help me with their Calendar API

<sub>[CAL-3640](https://linear.app/calcom/issue/CAL-3640/app-proton-calendar)</sub>

### PR DESCRIPTION
**Pull request by anikdhabal**
## What does this PR do?

- Fixes #14884 
- Fixes [CAL-3640](https://linear.app/calcom/issue/CAL-3640/app-proton-calendar)

## Mandatory Tasks (DO NOT REMOVE)

- [x] I have self-reviewed the code (A decent size PR without self-review might be rejected)
- [x] N/A I have added a Docs issue [here](https://github.com/calcom/docs/issues/new) if this PR makes changes that would require a [documentation change](https://docs.cal.com)
- [x] N/A I have added or modified automated tests that prove my fix is effective or that my feature works (PRs might be rejected if logical changes are not properly tested)

## How should this be tested?

- Are there environment variables that should be set?
- What are the minimal test data to have?
- What is expected (happy path) to have (input and output)?
- Any other important info that could help to test that PR

## Checklist

- I haven't read the [contributing guide](https://github.com/calcom/cal.com/blob/main/CONTRIBUTING.md)
- My code doesn't follow the style guidelines of this project
- I haven't commented my code, particularly in hard-to-understand areas
- I haven't checked if my changes generate no new warnings

### CODE REVIEW

**Comment by anikdhabal** [comment]
As of now, Proton Calendar does not have any public API, so in v1, let's make it dependent on Google Calendar. In v2, if we obtain the public API, then we can integrate it properly with OAuth.

**Comment by Ryukemeister** [comment]
> As of now, Proton Calendar does not have any public API, so in v1, let's make it dependent on Google Calendar. In v2, if we obtain the public API, then we can integrate it properly with OAuth.

@emrysal @ThyMinimalDev what do you folks think about this approach for v1?

**Comment by ThyMinimalDev** [comment]
not sure to understand the goal of this if Proton Calendar does not have any public API ?

**Comment by anikdhabal** [comment]
> not sure to understand the goal of this if Proton Calendar does not have any public API ?

Hey most third-party calendars offer a “subscribe to calendar” feature that allows to enter the shared calendar’s URL. Subscribing to a calendar in this way syncs the calendar to shared Proton Mail calendar.

In Google Calendar (web app), for example, go to Settings → Add calendar → From URL. Paste the shared calendar’s URL into the From URL field and click Add calendar.

And as you can see this calendar also dependent to google calendar. We have calendar apps like: vimcal, ammie etc. that is similar to this

**Comment by emrysal** [comment]
Although this works from a Calendar POV, Proton Calendar users are all about privacy and data ownership. Having this calendar work through Google makes little sense in my opinion because Proton users will never agree to this.

IMO this is not the way to implement - let's wait until Proton has a native way of calendar integration such as CalDav or at least some way native to us so we don't have to rely on Google for these users.

**Comment by anikdhabal** [comment]
> Although this works from a Calendar POV, Proton Calendar users are all about privacy and data ownership. Having this calendar work through Google makes little sense in my opinion because Proton users will never agree to this.
> 
> IMO this is not the way to implement - let's wait until Proton has a native way of calendar integration such as CalDav or at least some way native to us so we don't have to rely on Google for these users.

Yeah, this makes sense. Make this PR a draft until we have some implementation details

**Comment by Amit91848** [comment]
Since there is no definite timeline as to when things would progress, it's better to close this for now.

File: packages/app-store/proton-calendar/api/add.ts [hunk 1]
```diff
@@ -0,0 +1,20 @@
+import { createDefaultInstallation } from "@calcom/app-store/_utils/installation";
+import type { AppDeclarativeHandler } from "@calcom/types/AppHandler";
+
+import appConfig from "../config.json";
+
+const handler: AppDeclarativeHandler = {
+  appType: appConfig.type,
+  variant: appConfig.variant,
+  slug: appConfig.slug,
```
Comment (anikdhabal) [inline packages/app-store/proton-calendar/api/add.ts:9]: Pls fnuvnurvburv

**Pull request closed without merging**